Migration trial: Adapt import-focused flow connecting new Migration trial and Email verification steps

bogiii opened the pull request:
Related to https://github.com/Automattic/wp-calypso/issues/79915

## Proposed Changes

* Adapt **import-focused** flow connecting new **Migration trial** and **Email verification** steps.
* Transition from Simple to Atomic logic is moved from the `Upgrade` to `Migration Trial` step

**NOTE:** For easier code review, follow the commits.


## Testing Instructions

**Verified account test case:**
* Go to `/setup/import-focused?siteSlug={SIMPLE_SITE_SLUG}`
* Enter a self-hosted website URL (Jurassic Ninja)
* [Import preview step]: press the `Import your content` button
* [Upgrade plan step]: press the `Try it for free` button
* [Migration trial step]: press the `Start the trial and migrate`
* Check if the button is in a busy state
* After some time, check if the progress screen is rendered

**Non-verified account test case**
* Pick an existing JN site
* Make JP connection with non verified account
* Go to `/setup/import-focused?siteSlug={SIMPLE_SITE_SLUG}`
* Enter a JN URL
* [Import preview step]: press the `Import your content` button
* [Upgrade plan step]: press the `Try it for free` button
* [Migration trial step]: press the `Start the trial and migrate`
* Check if the Email verification step is rendered
* Press verify email CTA
* Go to your email box and verify your email
* Check if the Email verification step disappeared and the migration trial step is rendered
* [Migration trial step]: press the `Start the trial and migrate`
* Check if the button is in a busy state
* After some time, check if the progress screen is rendered

**site-setup flow**
* Go to `/setup/site-setup/import?siteSlug={SIMPLE_SITE_SLUG}`
* The rest is the same as the first testing scenario

**import-hosted-site flow**
* Go to `/setup/import-hosted-site/import`
* The rest is the same as the first testing scenario

![Screen Capture on 2023-08-02 at 19-29-45](https://github.com/Automattic/wp-calypso/assets/1241413/9f1c1a29-185b-4ad2-aa3a-3e9fc0b8762f)

**UPDATE:**

- We decided not to run the migration immediately, so after the `Start the trial and migrate` CTA, we are going to show an additional step with the option to populate SSH credentials to speed up the migration process.
<img width="679" alt="Screenshot 2023-08-03 at 12 39 35" src="https://github.com/Automattic/wp-calypso/assets/1241413/a87e00f8-894e-47ab-a625-03e92be53d48">


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

bogiii committed:
Update CTA text to match a proposed design

bogiii committed:
Import flow: connect migration trial and verify email steps

bogiii committed:
Import flow: provide "run" param which initiate migration process

bogiii committed:
Integrate Migration trial step into import flow

bogiii committed:
Remove default "run" flag from the URL which initiate immediate migration

ouikhuan commented on client/landing/stepper/declarative-flow/internals/steps-repository/migration-trial/style.scss:
```diff
@@ -1,22 +1,109 @@
+		flex-direction: column;
+		border: solid var(--studio-gray-5) 1px;
+		border-radius: 4px;
+		padding: 1.75em;
```
Out of curiosity, is there any reasons we're using `em` here instead of rem?

bogiii replied:
This change was not part of this PR, since I branched out from the **add/migration-trial** branch. Now it's rebased with the most recent version of trunk.

Anyway, you are right; there is not reason to be `em`. I'll replace it with a `rem` unit. It was copy/paste from one existing part of the CSS.

ouikhuan reviewed (commented):
Overall it works great! The steps we added here in the `import-flow.ts` should also be added to the `site-setup` and `import-hosted-site` flow as well, unless we decide not to show the free trial in those flows.

bogiii committed:
Remove `run` query param
We decided to not run the migration immediately and to show a step with an option for populating SSH credentials.

bogiii committed:
import-hosted-site flow: Register new import steps

bogiii committed:
site-setup-flow: Register new import steps

bogiii commented:
> should also be added to the site-setup and import-hosted-site flow as well

`site-setup` and `import-hosted-site` flows has been updated with the new steps.

ouikhuan commented on client/blocks/importer/wordpress/import-everything/pre-migration/index.tsx:
```diff
@@ -135,9 +138,10 @@ export const PreMigrationScreen: React.FunctionComponent< PreMigrationProps > =
-		addHostingTrial( targetSite.ID, PLAN_MIGRATION_TRIAL_MONTHLY );
-	};
+	// Initiate the migration if initImportRun is set
+	useEffect( () => {
+		initImportRun && startImport( { type: 'without-credentials' } );
+	}, [] );
```
Should we also remove this bit?

bogiii replied:
This feature is not in use right now. If we decide in the meantime to include it, it will be accessible by providing a `run` query param. So I'm leaving it as it is.

ouikhuan reviewed (approved):
LGTM 🚢
